packer/plugin.go

```diff
@@ -145,6 +146,12 @@ func (c *PluginConfig) DiscoverMultiPlugin(pluginName, pluginPath string) error
 		c.Builders.Set(key, func() (packersdk.Builder, error) {
 			return c.Client(pluginPath, "start", "builder", builderName).Builder()
 		})
+		PluginsDetailsStorage[fmt.Sprintf("%q-%q", PluginComponentBuilder, key)] = PluginDetails{
```

Comment:
The `PluginDetails` struct initialisation is a bit redundant, I believe we can build it once and reference it for each component we register? This doesn't change much in terms of performance to be clear (performance isn't a concern at all for this), but let's avoid this repetition if we can

Reply:
This metadata will grow over time so I agree whatever we can do to reduce the memory fingerprint the better.

Reply:
I don't think this will reduce the memory footprint in this case, if the structure is passed by value and not by reference, we'll still have multiple copies of the same structure with the current approach. My point was more that the structure we build is the same for each component we register, so we can simplify the registration code by initialising once instead of for each registration.

Reply:
I would argue that if memory footprint is a concern, we could probably only keep the path of the plugin, and run `describe` on it if needed, later in the process.

If we want to push it, we can get a couple of information from the name itself (arch, os, version, API version, and name can all be extracted), but it may be more prudent to rely on `describe` (only the version is checked at load-time, not the rest).

Reply:
> I don't think this will reduce the memory footprint in this case, if the structure is passed by value and not by reference, we'll still have multiple copies of the same structure with the current approach. 

That's fair. It might be a "we are not there yet thing" but I'm looking at it from the perspective of we will add a whole lot more metadata as we start to add integrations. We have to be cognizant of not duplicating all that metadata per build. But tbh I don't know how that will be stored yet so this idea of memory is probably moot. Especially, since Packer no longer bundles all the plugins, which were only getting bigger. I'll drop the memory discussion. 

But will agree on simplicity for the purpose of readability.